descqa/DeltaSigmaTest.py

```diff
@@ -0,0 +1,118 @@
+import os
+import numpy as np
+from scipy.interpolate import interp1d
+from astropy import units as u
+from astropy.coordinates import SkyCoord, search_around_sky
+import astropy.constants as cst
+from astropy.cosmology import WMAP7 # pylint: disable=no-name-in-module
+from .base import BaseValidationTest, TestResult
+from .plotting import plt
+
+__all__ = ['DeltaSigmaTest']
+
+class DeltaSigmaTest(BaseValidationTest):
+    """
+    This validation test looks at galaxy-shear correlations by comparing Delta
+    Sigma to the Singh et al (2015) measurements on the SDSS LOWZ sample.
+    """
+
+    def __init__(self, **kwargs):
+        # pylint: disable=super-init-not-called
+
+        # validation data
+        validation_filepath = os.path.join(self.data_dir, kwargs['data_filename'])
+        zmax = kwargs['zmax']
+
+        self.validation_data = np.loadtxt(validation_filepath)
+
+        # Create interpolation tables for efficient computation of sigma crit
+        z = np.linspace(0, zmax, zmax*100)
+        d1 = WMAP7.angular_diameter_distance(z) # in Mpc
+        self.angular_diameter_distance = interp1d(z, d1, kind='quadratic')
+
+        d2 = WMAP7.comoving_transverse_distance(z) # in Mpc
+        self.comoving_transverse_distance = interp1d(z, d2, kind='quadratic')
+
+    def run_on_single_catalog(self, catalog_instance, catalog_name, output_dir):
+        # pylint: disable=no-member
+
+        res = catalog_instance.get_quantities(['redshift_true', 'ra', 'dec', 'shear_1', 'shear_2',
+                                               'convergence', 'mag_true_i_sdss', 'mag_true_z_sdss',
+                                               'mag_true_g_sdss', 'mag_true_r_sdss'])
+
+        # Compute mask for lowz sample
```

Comment:
For future code maintainability, I recommend noting in the comment that this comes from section 3 of the BOSS target selection paper (https://arxiv.org/pdf/1509.06529.pdf) - people may not be familiar with what `lowz' means.

Reply:
added